src/modules/recipes/CreateNewRecipe/index.jsx

```diff
@@ -0,0 +1,217 @@
+import { ProtectedRoute } from "../../../ProtectedRoute";
+import { useForm } from "react-hook-form";
+import { useRecipeDispatch } from "../../../RecipeContext";
+import "./styles.css";
+import RecipeForm from "../RecipeForm";
+
+export const CreateNewRecipe = () => {
+  console.log("CreateNewRecipe rendered");
+
+  const dispatch = useRecipeDispatch();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      strMeal: "",
+      strCategory: "Beef",
+      strArea: "",
+      strInstructions: "",
+      ingredient1: "",
+      ingredient2: "",
+    },
+  });
+
+  const onSubmit = (data) => {
+    const recipeData = {
```

Comment:
idMeal should not be submitted from the form, it's something that is added on the server side after submitting data or in the RecipeContext reducer in our case.
Make sure that other values from exmample here are available in this form: https://www.themealdb.com/api/json/v1/1/search.php?s=Arrabiata

I can see that strIngredient3-20 and strMeasure1-20 and some other values are missing.

Reply:
@selceli you're still not adding idMeal attribute to the data which results in the recipe details page not being displayed correctly.